### ISSUE TITLE
single-table inheritance mechanism failed to locate subclass

### ISSUE THREAD

**Issue by holcomb1977** [rails/rails#45812]
Received the following error twice:

The single-table inheritance mechanism failed to locate the subclass: 'ForemanOpenscap::ComplianceStatus.' This error is raised because the column 'type' is reserved for storing the class in case of inheritance. Please rename this column if you didn't intend to be used for storing the inheritance class or overwrite HostStatus::Status.inheritance_column to use another column for that information.  

I am not a linux guy, so any assistance on this error...in layman's terms...would be greatly appreciated.

### Steps to reproduce

```ruby
# Your reproduction script goes here
```

### System configuration
**Rails version**:

**Ruby version**:

**Comment by eileencodes** [comment]
hey @holcomb1977 this isn't a linux issue, it's an error being thrown by your Rails application. Based on the subclass name (`ForemanOpenscap::ComplianceStatus`) it looks like you are using or have in the past used this plugin https://github.com/theforeman/foreman_openscap. This sounds like either there is a bug in the plugin (which we don't control or work on) or a bug in how your application is interacting with the plugin. While the error comes from the Rails framework, this isn't a bug in the framework but rather an error about an incorrect setup.

The error is saying that you have a `type` column on a model that is set up for STI or single-table inheritance, but that Rails can't find the subclass (in this case `ForemanOpenscap::ComplianceStatus`) that is stored in that column in your database. Documentation on STI with Rails can be found here https://guides.rubyonrails.org/association_basics.html#single-table-inheritance-sti.

Since this isn't a bug in the Rails framework and is either a bug in the plugin or your application I'm going to close this. Hope you find the solution soon!

**Comment by holcomb1977** [comment]
thanks. do you think it could be resolved by upgrading from our current Redhat satellite to Redhat satellite insights?

**Comment by eileencodes** [comment]
> thanks. do you think it could be resolved by upgrading from our current Redhat satellite to Redhat satellite insights?

I don't think so unless Redhat Satellite Insights is what's providing the plugin or code that's throwing this error. You'll have to ask their support if that's the case.

**Comment by holcomb1977** [comment]
roger. Thank you @eileencodes
